src/NuGetForUnity/Editor/PackageCacheManager.cs

```diff
@@ -156,5 +153,24 @@ private static INugetPackage GetInstalledPackage([NotNull] INugetPackageIdentifi
                 installedPackage.Version);
             return installedPackage;
         }
+
+        private static string DetermineCacheOutputDirectory()
+        {
+            const string cacheEnvironmentVariableName = "NuGetCachePath";
+            var cacheFromEnvironment = Environment.GetEnvironmentVariable(cacheEnvironmentVariableName);
+            if (!string.IsNullOrEmpty(cacheFromEnvironment))
```

Comment:
In case the path given in env doesn't exist should we try to CreateDirectory or just fallback to default handling bellow? Although we might want to add some handling if default path doesn't exist as well.

Reply:
`Directory.CreateDirectory(CacheOutputDirectory);` is called inside the static constructor for both cases.
I think this is the expected behavior.
Poth path's / the name of the environment variable are the once used by the official NuGet client prior to 3.5. Eventually we can switch to the newer cache format / location later: [docu](https://learn.microsoft.com/en-us/nuget/consume-packages/managing-the-global-packages-and-cache-folders)